rdbms/src/main/java/org/spine3/server/storage/jdbc/JdbcStandStorage.java

```diff
@@ -64,73 +67,121 @@ public Object apply(@Nullable AggregateStateId input) {
         }
     };
 
+    private static final Function<Map.Entry<AggregateStateId, EntityRecord>, Map.Entry<Object, EntityRecord>> MAP_TARNSFORMER
+            = new Function<Map.Entry<AggregateStateId, EntityRecord>, Map.Entry<Object, EntityRecord>>() {
+        @Override
+        public Map.Entry<Object, EntityRecord> apply(
+                @Nullable Map.Entry<AggregateStateId, EntityRecord> entry) {
+            checkNotNull(entry);
+            final Object newKey = entry.getKey()
+                                       .getAggregateId();
+            final Map.Entry<Object, EntityRecord> newEntry = new SimpleEntry<>(
+                    newKey,
+                    entry.getValue());
+            return newEntry;
+        }
+    };
+
     @SuppressWarnings("unchecked")
     protected JdbcStandStorage(Builder builder) {
         super(builder.isMultitenant);
         recordStorage = JdbcRecordStorage.newInstance(
                 checkNotNull(builder.dataSource),
                 builder.isMultitenant,
-                checkNotNull(builder.entityStorageQueryFactory),
-                checkNotNull(builder.stateDescriptor));
+                checkNotNull(builder.recordStorageQueryFactory));
     }
 
     @Override
-    public ImmutableCollection<EntityStorageRecord> readAllByType(TypeUrl type) {
+    public ImmutableCollection<EntityRecord> readAllByType(TypeUrl type) {
         return readAllByType(type, FieldMask.getDefaultInstance());
     }
 
     @Override
-    public ImmutableCollection<EntityStorageRecord> readAllByType(final TypeUrl type, FieldMask fieldMask) {
-        final Map<AggregateStateId, EntityStorageRecord> allRecords = readAll(fieldMask);
-        final Map<AggregateStateId, EntityStorageRecord> resultMap = Maps.filterKeys(allRecords, new Predicate<AggregateStateId>() {
-            @Override
-            public boolean apply(@Nullable AggregateStateId stateId) {
-                checkNotNull(stateId);
-                final boolean typeMatches = stateId.getStateType()
-                        .equals(type);
-                return typeMatches;
-            }
-        });
+    public ImmutableCollection<EntityRecord> readAllByType(final TypeUrl type,
```

Comment:
Isn't it possible to achieve the same by executing a database query? 

If you don't have enough details stored in the database, let's discuss first.